### PR TITLE
Update to use openapi-core 0.19.0

### PR DESCRIPTION
**Pull request by miketheman**
This is an incomplete PR, but I wanted to share where I've gotten to so far.
A lot of the foundations are based on work done in #206 

This includes the assumption of remaining OpenAPI 3.0 compliant (which we can absolutely change going forward), I wanted to get as close as I could to pyramid_openapi3 0.16 before doing so - and get some clean test passes.

I had to skip a few tests to complete the pass, which I consider insufficient for merge, so I've left some TODO and FIXME inline with some notes, and would appreciate any pointers to help resolve.

I tried to split commits out to show the work done at each step, hopefully that will make reviewing easier.

### CODE REVIEW

**Comment by miketheman** [comment]
Side note - should `Blueracer.io` be removed from CI and also the branch protection requirements? The service has been discontinued.

**Comment by zupo** [comment]
> Side note - should `Blueracer.io` be removed from CI and also the branch protection requirements? The service has been discontinued.

Yes, please.

File: pyramid_openapi3/tests/test_extract_errors.py [hunk 1]
```diff
@@ -126,11 +125,13 @@ def test_invalid_path_parameter(self) -> None:
...
         res = self._testapp(
             view=self.foo, endpoints=endpoints, route="/foo/{bar}"
-        ).post("/foo/bar", status=400)
+        ).post("/foo/not_a_number", status=400)
         assert res.json == [
             {
-                "exception": "CastError",
-                "message": "Failed to cast value to integer type: bar",
+                # TODO: used to be CastError, might have been fixed later
+                "exception": "ParameterValidationError",
```
Comment (zupo) [inline pyramid_openapi3/tests/test_extract_errors.py:132]: This error is fine.

**Comment by zupo** [comment]
@miketheman: this looks much better than I hoped!

File: pyramid_openapi3/tests/test_extract_errors.py [hunk 2]
```diff
@@ -558,34 +560,34 @@ def foo(*args: t.Any) -> Exception:
         ]
 
 
+@unittest.skip("Needs to be re-implemented")
```
Comment (zupo) [inline pyramid_openapi3/tests/test_extract_errors.py:563]: This one definitely needs to be included in the PR.

**Review by miketheman — COMMENTED** [review]
I've got everything passing `make all`.

I'm not super familiar with nix-shell or how to update the stack correctly - I spent a few hours trying to figure it out, but ended up more confused than when I started.

File: pyramid_openapi3/tests/test_contenttypes.py [hunk 3]
```diff
@@ -72,7 +72,10 @@ def test_post_json(self) -> None:
         res = self._testapp().post_json("/foo", {"bar": "baz"}, status=200)
         self.assertEqual(res.json, {"bar": "zab"})
 
-    def test_post_form(self) -> None:
+    # FIXME: Something deep in request validation isn't happy right now.
+    # Might be related: https://github.com/Pylons/pyramid_openapi3/issues/199
+    @unittest.expectedFailure
```
[the remark this answers is not in the document]
Comment (zupo) [reply pyramid_openapi3/tests/test_contenttypes.py:77]: I'll have a look.

[the remark this answers is not in the document]
Comment (miketheman) [reply pyramid_openapi3/tests/test_contenttypes.py:77]: Hi @zupo ! Have you had a chance to come back to this, or the nix-shell updates necessary?

[the remark this answers is not in the document]
Comment (zupo) [reply pyramid_openapi3/tests/test_contenttypes.py:77]: No, sorry, I haven't gotten to it yet, busy days at work.

**Comment by zupo** [comment]
@miketheman: can you give me write permissions to your fork, so I can push CI/nix fixes?

**Comment by miketheman** [comment]
> @miketheman: can you give me write permissions to your fork, so I can push CI/nix fixes?
> 
> Also, could you rebase it off of latest `main` to get all the recent fixes in.

Rebased, and invitation sent!

**Comment by zupo** [comment]
@miketheman: CI found some failing tests in the examples/ folder. I fixed one, but ran out of time to do the rest, gotta run.

File: pyramid_openapi3/tests/test_contenttypes.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (miketheman) [reply pyramid_openapi3/tests/test_contenttypes.py:77]: @Wim-De-Clercq Thanks for the pointer, totally works.

[the remark this answers is not in the document]
Comment (miketheman) [reply pyramid_openapi3/tests/test_contenttypes.py]: @Wim-De-Clercq I added a commit, and attributed it to you - hope that's okay! [`04cbaa5` (#220)](https://github.com/Pylons/pyramid_openapi3/pull/220/commits/04cbaa5475afdce8b5320018345e77a8d0528a25)

**Comment by miketheman** [comment]
@zupo Our powers combined, it's working!  With the change to drop Python 3.8, please update the branch protection requirements to remove "Python 3.8 Tests".

**Pull request merged**